logger/modules/log_file_merger_helpers.py

```diff
@@ -0,0 +1,191 @@
+"""
+Log file merger helper functions.
+"""
+
+import pathlib
+from datetime import datetime
+
+from ..read_yaml.modules import read_yaml
+
+
+MERGED_LOGS_FILENAME = "merged_logs.log"
+LOG_FILE_SUFFIX = ".log"
+
+
+def read_configuration(
+    config_file_path: pathlib.Path,
+) -> "tuple[bool, pathlib.Path | None, str | None, str | None]":
+    """
+    Reads the configuration YAML file and returns the log directory path and file datetime format.
+
+    config_file_path: Path to the configuration YAML file.
+
+    Returns: Success, log directory path, file datetime format, log datetime format.
+    """
+    result, config = read_yaml.open_config(config_file_path)
+    if not result:
+        print(f"ERROR: Failed to load configuration file: {config_file_path}")
+        return False, None, None, None
+
+    # Extract log directory path and file datetime format from configuration dictionary
+    try:
+        log_directory_path_config = config["logger"]["directory_path"]
+        # Convert the directory_path string to a Path object
+        log_directory_path = pathlib.Path(log_directory_path_config)
+        file_datetime_format = config["logger"]["file_datetime_format"]
+        log_datetime_format = config["logger"]["log_datetime_format"]
+    except KeyError as exception:
+        print(f"Config key(s) not found: {exception}")
+        return False, None, None, None
+
+    if not log_directory_path.exists():
+        print(f"No log directory exists at: {log_directory_path}")
+        return False, None, None, None
+
+    return True, log_directory_path, file_datetime_format, log_datetime_format
+
+
+def get_log_run_directories(
+    log_directory_path: pathlib.Path, file_datetime_format: str, override: bool
+) -> "tuple[bool, list[pathlib.Path] | None]":
+    """
+    Retrieves the paths of log run directories.
+
+    log_directory_path: Path to the log directory.
+    file_datetime_format: Datetime format in the filename.
+    override: Overwrite merged_logs files.
```

Comment:
This is not what the parameter actually does, there is no overwriting involved in this function.